peps/pep-0785.rst

```diff
@@ -39,116 +39,127 @@ often write code to process or respond to individual leaf exceptions, for
 example when implementing middleware, error logging, or response handlers in
 a web framework.
 
-`Searching GitHub`__ found four implementations of :meth:`!leaf_exceptions` by
-various names in the first sixty hits, of which none handle
-tracebacks.\ [#numbers]_  The same search found thirteen cases where
-:meth:`!.leaf_exceptions` could be used.  We therefore believe that providing
+In April 2025, `searching GitHub`__ [#naming]_ found four implementations of
+:meth:`!leaf_exceptions` by various names in the first sixty hits, of which
+none handle tracebacks.\ [#numbers]_  The same search found thirteen cases
+where :meth:`!.leaf_exceptions` could be used.  In August, `searching`__
+for ``leaf_exceptions`` identified a total of five distinct implementations,
+none of which handled tracebacks, and three cases of copying an early
+reference implementation from this PEP.  We therefore believe that providing
 a method on the :class:`BaseException` type with proper traceback preservation
 will improve error-handling and debugging experiences across the ecosystem.
 
 __ https://github.com/search?q=%2Ffor+%5Cw%2B+in+%5Beg%5D%5Cw*%5C.exceptions%3A%2F+language%3APython&type=code
 
-The rise of exception groups has also made re-raising exceptions caught by an
-earlier handler much more common: for example, web-server middleware might
-unwrap ``HTTPException`` if that is the sole leaf of a group:
+__ https://github.com/search?q=%2Fdef+.*leaf_exceptions.*%5C%28%2F+language%3APython&type=code
+
+When Python raises an exception, it automatically sets ``__context__`` to
+track what was being handled at the time. This normally provides valuable
+debugging information. However, with ``except*``, this behavior can create
+a problem.  Consider this example middleware, which unwraps a single-exception
+group:
 
 .. code-block:: python
 
     except* HTTPException as group:
-        first, *rest = group.leaf_exceptions()  # get the whole traceback :-)
+        (exc, tb), *rest = group.leaf_exceptions()
         if not rest:
-            raise first
+            raise exc.with_traceback(tb)
         raise
 
-However, this innocent-seeming code has a problem: ``raise first`` will do
-``first.__context__ = group`` as a side effect. This discards the original
+In this case, the implicit ``exc.__context__ = group`` discards the original
 context of the error, which may contain crucial information to understand why
 the exception was raised. In many production apps it also causes tracebacks
```

Comment:
`leaf_exceptions()` loses the `__context__` of all `ExceptionGroups` on the path from the leaf to the root. Here you're fixing it only for the last `__context__`, but that's not enough.